plugins/govppmux/govpp_channel.go

```diff
@@ -0,0 +1,74 @@
+package govppmux
+
+import (
+	"time"
+
+	govppapi "git.fd.io/govpp.git/api"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging/logrus"
+)
+
+// goVppChan implements govpp channel interface. Instance is returned by NewAPIChannel() or NewAPIChannelBuffered(),
+// and contains *govpp.channel dynamic type (vppChan field). Implemented methods allow custom handling of low-level
+// govpp.
+type goVppChan struct {
+	govppapi.Channel
+	// Retry data
+	retry retryConfig
+}
+
+// govppRequestCtx is custom govpp RequestCtx.
+type govppRequestCtx struct {
+	// Original request context
+	requestCtx govppapi.RequestCtx
+	// Function allowing to re-send request in case it's granted by the config file
+	sendRequest func(govppapi.Message) govppapi.RequestCtx
+	// Parameter for sendRequest
+	requestMsg govppapi.Message
+	// Retry data
+	retry retryConfig
+}
+
+// helper struct holding info about retry configuration
+type retryConfig struct {
+	attempts int
+	timeout  time.Duration
+}
+
+// ReceiveReply handles request and returns error if occurred. Also does retry if this option is available.
+func (r *govppRequestCtx) ReceiveReply(reply govppapi.Message) error {
+	var timeout time.Duration
+	maxAttempts := r.retry.attempts
+	if r.retry.timeout > 0 { // Default value is 500ms
+		timeout = r.retry.timeout
+	}
+
+	var err error
+	// Receive reply from original send
+	if err = r.requestCtx.ReceiveReply(reply); err == core.ErrNotConnected && maxAttempts > 0 {
+		// Try to re-sent requests
+		for attemptIdx := 1; attemptIdx <= maxAttempts; attemptIdx++ {
+			// Wait, then try again
+			time.Sleep(timeout)
+			logrus.DefaultLogger().Warnf("Govppmux: retrying binary API message %v, attempt: %d",
+				r.requestMsg.GetMessageName(), attemptIdx)
+			if err = r.sendRequest(r.requestMsg).ReceiveReply(reply); err == core.ErrNotConnected {
+				continue
```

Comment:
I just just one more simplification 🙄

Reply:
Let's check if `err != ErrNotConnected` and then break the loop. Then you don't need return on line 58 at all, because the return on line 62 will return same error.